### PR TITLE
spi_prepare and spi_execute_plan for python and shared code.

### PR DESCRIPTION
**Pull request by paul-guo-**

### CODE REVIEW

**Comment by paul-guo-** [comment]
The theory for spi_prepare is that we save plan on QE and transfer the plan pointer to pyclient although the value is meaningless for pyclient. I'm adding test cases, but the code is generally ready for CR.

File: src/common/messages/message_raw.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ *------------------------------------------------------------------------------
+ */
+#ifndef PLC_MESSAGE_RAW_H
+#define PLC_MESSAGE_RAW_H
+
+#include "message_base.h"
+
+typedef struct plcMsgRaw {
+    base_message_content;
+	int   size;
```
Comment (zhangh43) [inline src/common/messages/message_raw.h]: format

**Comment by zhangh43** [comment]
+1

**Comment by HaozhouWang** [comment]
+1

**Comment by stanlyxiang** [comment]
+1

**Pull request merged**